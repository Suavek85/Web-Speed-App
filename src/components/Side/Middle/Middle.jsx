import React from 'react'
import { useSelector } from "react-redux"
import SideCard from '../SideCard'
import { RESULT } from '../../../constants/content.js'

export default function Middle() {

  const getDataState = useSelector(state => state.stateReducer.getData)

  const content = {
    header: 'Unused Javascript',
    mainContent: 'Remove unused JavaScript to reduce bytes consumed by network activity.',
    result: [RESULT],
  }

  const data = [getDataState[1]]
  const position = 'bottom'
  const props = { content, data, position }


  return (
    <SideCard {...props} />
  )
}