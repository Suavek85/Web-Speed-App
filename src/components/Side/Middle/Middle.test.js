import React from 'react'
import * as redux from 'react-redux'
import { shallow } from 'enzyme'
import Middle from './Middle'

describe('Middle snapshot', () => {

  it('should match the snapshot', () => {
    const mockedResponse = [0.67, 0.47, "https://www.bbc.com/", "SLOW", "AVERAGE", "FAST", 0.98]
    const spyUseSelector = jest.spyOn(redux, 'useSelector')
    spyUseSelector.mockReturnValue(mockedResponse)
    const wrapper = shallow(<Middle />)
    expect(wrapper).toMatchSnapshot()
  })

})