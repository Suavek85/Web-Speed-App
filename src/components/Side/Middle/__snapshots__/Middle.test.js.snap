// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Middle snapshot should match the snapshot 1`] = `
<Section
  content={
    Object {
      "header": "Unused Javascript",
      "mainContent": "Remove unused JavaScript to reduce bytes consumed by network activity.",
      "result": Array [
        "Result: ",
      ],
    }
  }
  data={
    Array [
      0.47,
    ]
  }
  position="bottom"
/>
`;
